frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideInBranchBelowAction.java

```diff
@@ -0,0 +1,198 @@
+    if (localBranch == null) {
+      var createNewBranchDialogBranchName = createOrCheckoutNewBranch(project, selectedVcsRepository, parentName,
+          slideInDialogBranchName);
+      if (!slideInDialogBranchName.equals(createNewBranchDialogBranchName)) {
+        createNewBranchDialogBranchName = createNewBranchDialogBranchName != null
+            ? createNewBranchDialogBranchName
+            : "no name provided";
+        notifier.notifyWeakError(
+            GitMacheteBundle.message("action.GitMachete.BaseSlideInBranchBelowAction.notification.fail.mismatched-names",
+                slideInDialogBranchName,
+                createNewBranchDialogBranchName));
+        return;
+      }
+    }
+
+    // TODO (#430): expose getParent from branch layout api
+    var parentEntry = branchLayout.findEntryByName(parentName);
+    var entryAlreadyExistsBelowGivenParent = parentEntry
+        .map(entry -> entry.getChildren())
+        .map(children -> children.map(e -> e.getName()))
+        .map(names -> names.contains(slideInDialogBranchName))
+        .getOrElse(false);
+
+    if (entryAlreadyExistsBelowGivenParent) {
+      log().debug("Skipping action: Branch layout entry already exists below given parent");
+      return;
+    }
+
+    new Task.Backgroundable(project, GitMacheteBundle.message("action.GitMachete.BaseSlideInBranchBelowAction.task-title")) {
+
+      @Override
+      public void run(ProgressIndicator indicator) {
+        Path macheteFilePath = getMacheteFilePath(selectedVcsRepository);
+
+        var entryToSlideIn = branchLayout.findEntryByName(slideInDialogBranchName)
+            .getOrElse(
+                new BranchLayoutEntry(slideInDialogBranchName, /* customAnnotation */ null, /* children */ List.empty()));
+
+        var newBranchLayout = Try
+            .of(() -> branchLayout.slideIn(parentName, entryToSlideIn))
+            .onFailure(
+                t -> notifier.notifyError(
+                    /* title */ GitMacheteBundle.message("action.GitMachete.BaseSlideInBranchBelowAction.notification.fail",
+                        slideInDialogBranchName),
+                    getMessageOrEmpty(t)))
+            .toOption();
+
+        newBranchLayout.map(nbl -> Try.run(() -> branchLayoutWriter.write(macheteFilePath, nbl, /* backupOldLayout */ true))
+            .onFailure(t -> notifier.notifyError(
+                /* title */ GitMacheteBundle
+                    .message("action.GitMachete.BaseSlideInBranchBelowAction.notification.fail.branch-layout-write"),
+                getMessageOrEmpty(t))));
+      }
+    }.queue();
+  }
+
+  private static String getMessageOrEmpty(Throwable t) {
+    return t.getMessage() != null ? t.getMessage() : "";
+  }
+
+  @Nullable
+  String createOrCheckoutNewBranch(Project project, GitRepository gitRepository, String startPoint, String initialName) {
+    var repositories = List.of(gitRepository).asJava();
```

Comment:
Nit: `new java.util.ArrayList(gitRepository)`? (or whatever the c'tor/factory method is)